js/src/main/scala/crystal/react/reuse/package.scala

```diff
@@ -126,4 +142,122 @@ package object reuse {
     ): Reuse[B] = Reuse.currying(r, s, t).in(fn)
   }
 
+  implicit class ReuseViewF[F[_]: Monad, A](val rv: Reuse[ViewF[F, A]]) {
+    val get: A                                           = rv.value.get
+    val modCB: ((A => A), A => F[Unit]) => F[Unit]       = rv.value.modCB
+    def modAndGet(f: A => A)(implicit F: Async[F]): F[A] = rv.value.modAndGet(f)
+
+    def zoom[B](getB: A => B)(modB: (B => B) => A => A): Reuse[ViewF[F, B]] =
+      rv.map(_.zoom(getB)(modB))
+
+    def zoomOpt[B](getB: A => Option[B])(modB: (B => B) => A => A): Reuse[ViewOptF[F, B]] =
+      rv.map(_.zoomOpt(getB)(modB))
+
+    def zoomList[B](
+      getB: A => List[B]
+    )(modB: (B => B) => A => A): Reuse[ViewListF[F, B]] = rv.map(_.zoomList(getB)(modB))
+
+    def as[B](iso: Iso[A, B]): Reuse[ViewF[F, B]] = zoom(iso.asLens)
+
+    def asOpt: Reuse[ViewOptF[F, A]] = zoom(Iso.id[A].asOptional)
+
+    def asList: Reuse[ViewListF[F, A]] = zoom(Iso.id[A].asTraversal)
+
+    def zoom[B](lens: Lens[A, B]): Reuse[ViewF[F, B]] = zoom(lens.get _)(lens.modify)
+
+    def zoom[B](optional: Optional[A, B]): Reuse[ViewOptF[F, B]] =
+      zoomOpt(optional.getOption _)(optional.modify)
+
+    def zoom[B](prism: Prism[A, B]): Reuse[ViewOptF[F, B]] =
+      zoomOpt(prism.getOption _)(prism.modify)
+
+    def zoom[B](traversal: Traversal[A, B]): Reuse[ViewListF[F, B]] =
+      zoomList(traversal.getAll _)(traversal.modify)
+
+    def withOnMod(f: A => F[Unit]): Reuse[ViewF[F, A]] = rv.map(_.withOnMod(f))
+
+    def widen[B >: A]: Reuse[ViewF[F, B]] = rv.map(_.widen[B])
+
+    def unsafeNarrow[B <: A]: Reuse[ViewF[F, B]] =
+      zoom(_.asInstanceOf[B])(modB => a => modB(a.asInstanceOf[B]))
+
+    def to[F1[_]: Monad](
+      toF1:   F[Unit] => F1[Unit],
+      fromF1: F1[Unit] => F[Unit]
+    ): Reuse[ViewF[F1, A]] = rv.map(_.to[F1](toF1, fromF1))
+
+    def mapValue[B, C](f: ViewF[F, B] => C)(implicit ev: A =:= Option[B]): Option[C] =
```

Comment:
Maybe `f` can be `Reuse[ViewF[F, B]] => C` so we can propagate reusability.

Reply:
Hmm. I neglected to change the return type on this one (I did in ReuseViewOptF]). The return really is `Reuse[Option[C]]`, but the automatic conversion is changing it. Do you want `Reuse` in the `f` or just the return?

Reply:
If `f` is `Reuse[ViewF[F, B]] => C`, would we combine the reusability of `f` with the current reusability?

Reply:
I think that what's important is that we pass a `Reuse[View[A]]` into the function. We should build that with the same reusability that we have for the `Reuse[View[Option[A]]]`. We can return `Option[C]`.

Use case for this is for example:

``` scala
val view: View[ADTRoot]

<.div(
  view
    .zoom(ADTRoot.prismToOneOfTheSubtypes)
    .mapValue(stillReusableView => Component(stillReusableView))
    .whenDefined
)
```

Reply:
Ah-ha. I was tired and my brain didn't really grok what the method did - I just did something mechanical.